### PR TITLE
Option to print tensor data in hex format

### PR DESCRIPTION
**Pull request by AlexBrownAMD**

### CODE REVIEW

**Review by benjaminulmer — APPROVED** [review]
The template specializations for float and double are annoying, but I can't think of a better way of doing it. In other words, looks good.

File: Tensile/Common.py
Comment (cgmillette) [inline Tensile/Common.py:182]: Just curious about not using boolean values here. Below, booleans are used.

File: Tensile/Source/lib/include/Tensile/TensorDescriptor.hpp
Comment (cgmillette) [inline Tensile/Source/lib/include/Tensile/TensorDescriptor.hpp:347]: Might be able to write a smaller function to do the stream treatments:
template<typename T>
inline void WriteElement(std::ostream& stream, T* data, int index){  stream << data[index]; }

template<>
inline void WriteElement<float>(std::ostream& stream, float* data, int index){  stream << *reinterpret_cast<const uint32_t*>(&data[i]); }

...

And then plug this into the WriteTensor1DHex function to avoid specializing this bigger function and copying loops.

File: Tensile/Source/lib/include/Tensile/TensorDescriptor.hpp
Comment (cgmillette) [inline Tensile/Source/lib/include/Tensile/TensorDescriptor.hpp:409]: Same as above. Should be able to solve a lot of the repetition.

**Review by babakpst — DISMISSED** [review]
Blocked until after FC merge.

**Review by babakpst — CHANGES_REQUESTED** [review]
Blocked until FC merges are complete.

**Comment by babakpst** [comment]
Closing this PR, as this is not a priority now.

**Pull request closed without merging**
